src/main/java/mate/hwdao/dao/impl/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,197 @@
+import mate.hwdao.model.Driver;
+import mate.hwdao.model.Manufacturer;
+import mate.hwdao.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (manufacturer_id, model) VALUES (?, ?)";
+        try (PreparedStatement preparedStatement
+                     = ConnectionUtil.getConnection()
+                .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, car.getManufacturer().getId());
+            preparedStatement.setString(2, car.getModel());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            while (resultSet.next()) {
+                car.setId(resultSet.getObject("GENERATED_KEY", Long.class));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't save " + car
+                    + " failed", ex);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT * FROM cars "
+                + "INNER JOIN manufactures ON cars.manufacturer_id = manufactures.id "
+                + "WHERE cars.id = ? and deleted = false;";
+        try (PreparedStatement preparedStatement
+                     = ConnectionUtil.getConnection()
+                .prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            Car car = null;
+            while (resultSet.next()) {
+                car = makeCar(resultSet);
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get car with " + id, ex);
+        }
+
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars INNER JOIN manufactures"
+                + " ON cars.manufacturer_id = manufactures.id WHERE deleted = false";
+        try (PreparedStatement preparedStatement
+                     = ConnectionUtil.getConnection()
+                .prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(makeCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get list of cars ", ex);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCar = "UPDATE cars SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND deleted = false";
+        String deletedOldDrivers = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String setNewDrivers = "INSERT INTO cars_drivers(driver_id, car_id) VALUES (?, ?)";
```

Comment:
create 2 private methods, each opening its own connection, and call them after the try-catch block

Reply:
done